### ISSUE TITLE
Building on Windows

### ISSUE THREAD

**Issue by peterkronenberg** [kaldi-asr/kaldi#4445]
Sorry if this isn't the right place for this.  Wondering if there is any progress on building Kaldi on Windows?  I'm want to be able to use Vosk and following the general procedure they have at https://alphacephei.com/vosk/install.  Not having much luck.  I've tried doing it under Msys.  

Searching the web, it seems that many people have attempted this, but haven't seen anyone yet who has it working.  Any suggestions?  I'll try any environment or tool on Windows that I need to.  I'm an experienced Java developer, but don't know much of the nitty-gritty around gcc and associated tools on Linux (although I can find my way around a Linux system)

Any help would be appreciated

**Comment by danpovey** [comment]
Depends what you need Kaldi for.  We do have some resources for building on
Windows- see windows/-- but not all parts of Kaldi build.
It's possible to get it working for inference (although you'll have to do
some messing around with BLAS libraries probably), but not really for
training.
It's hard to keep things working on Windows because they change the format
of the build files too often.

**Comment by GeorgeS2019** [comment]
@danpovey  is there a plan to develop a .NETcore wrapper for c# developer. Currently I am using .NETCore DeepSpeech. However, it seems kaldi for german is making more progress than deepspeech for German. I urge Kaldi to consider a .NETcore wrapper.

**Comment by danpovey** [comment]
I'm afraid not.  Right now I am focusing on the "next generation" of Kaldi,
which is separate projects (k2/lhotse/snowfall)...

**Comment by GeorgeS2019** [comment]
@danpovey  after further effort, just realized how to do that using dotnetCore.  Great work for Kaldi!

**Comment by nshmyrev** [comment]
Since this issue has higher visibility worth to note that we successfully build Kaldi for Win32 and Win64 with MINGW, the docker for build is here:

https://github.com/alphacep/vosk-api/blob/master/travis/Dockerfile.win

the prebuilt binaries for dotnet are available here for example:

https://www.nuget.org/packages/Vosk/

Windows and Linux are supported, probably Mac too.

**Comment by GeorgeS2019** [comment]
@nshmyrev  **[Kaldi Visual Studio Solution](https://github.com/Smorodov/kaldi_vosk_win_cmake)** generated by cmake

**Comment by marekCs** [comment]
> @danpovey after further effort, just realized how to do that using dotnetCore. Great work for Kaldi!

Hey mate, can you please share info or at least the source where you figured out how to set up everything for .NET please?

**Comment by csukuangfj** [comment]
@marekCs 

Will you consider Next-gen Kaldi?

Both [sherpa-onnx][sherpa-onnx] and [sherpa-ncnn][sherpa-ncnn] support .Net, tested on Linux, macOS, and Windows.

[sherpa-onnx]: https://github.com/k2-fsa/sherpa-onnx
[sherpa-ncnn]: https://github.com/k2-fsa/sherpa-ncnn

**Comment by marekCs** [comment]
Wow, will check it all thx.

Sent from Proton Mail for iOS

V po, zář 4, 2023 v 08:20, Fangjun Kuang ***@***.***(mailto:V po, zář 4, 2023 v 08:20, Fangjun Kuang <<a href=)> napsal/a:

**Comment by csukuangfj** [comment]
Please see the documentation at
https://k2-fsa.github.io/sherpa/onnx/csharp-api/index.html

**Comment by marekCs** [comment]
Thank you so much. Just for clarifying. Streaming/unstreamed (online/offline) doesn't depend on me using it on an INTERNET online/offline machine, right? It's a live voice or already stored recorded voice, am I right?

Sent from Proton Mail for iOS

V po, zář 4, 2023 v 08:33, Fangjun Kuang ***@***.***(mailto:V po, zář 4, 2023 v 08:33, Fangjun Kuang <<a href=)> napsal/a:

**Comment by csukuangfj** [comment]
> Streaming/unstreamed (online/offline) doesn't depend on me using it on an INTERNET online/offline machine

Yes, you are right. online/offline has nothing to do with computer networks here.

> It's a live voice or already stored recorded voice, am I right?

Yes, you are right.
